lib/experimental/fonts-api/bc-layer/class-gutenberg-fonts-api-bc-layer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Fonts API BC Layer helpers.
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ * This file is now part of the API's Backwards-Compatibility (BC) layer.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts API
+ * @since      15.7.0
+ */
+
+/**
+ * Class Gutenberg_Fonts_API_BC_Layer
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ */
+class Gutenberg_Fonts_API_BC_Layer {
+
+	/**
+	 * Determines if the given fonts array is the deprecated array structure.
+	 *
+	 * @param array $fonts Array of fonts to check.
+	 * @return bool True when deprecated structure, else false.
+	 */
+	public static function is_deprecated_structure( array $fonts ) {
+		// Checks the first key to determine if it's empty or non-string.
+		foreach ( array_keys( $fonts ) as $font_family ) {
+			return ! WP_Fonts_Utils::is_defined( $font_family );
+		}
```

Comment:
I'm wondering if a polyfill can be used here:
```suggestion
		return ! WP_Fonts_Utils::is_defined( array_key_first ( $fonts ) );
```

If that's not an option, I propose to avoid calling `array_keys()` to improve performance:

```php
		foreach ( $fonts as $font_family => $font ) {
			return ! WP_Fonts_Utils::is_defined( $font_family );
		}
		
		return false;
```

Reply:
Good catch @anton-vlasenko. `array_key_first()` polyfill was introduced in WP 5.9.0. So yes, it can and should be used.

Reply:
Hmm the more I think about this, @anton-vlasenko I think you might be on to something. Let's change this code to check _all_ first dimension keys. The first it finds that is not defined, then return true. Updating now.

Reply:
Done here https://github.com/WordPress/gutenberg/pull/50077/commits/12db6074a9825cc58f7c68f8999428091fcce0df